Timeline component overhaul (#5258)
SkylineCommunications/dataminer-docs: dataminer/Functions/Dashboards_and_Low_Code_Apps/Visualizations/Available_visualizations/Other/Timeline_component.md

> Prior to DataMiner 10.3.0 CU14/10.4.0 CU2/10.4.5, highlights can be configured using the *Highlight range* setting. This setting is still available in later DataMiner versions and can be used in combination with highlights set by actions. Its behavior remains the same: a highlight set by the *Highlight range* setting will not expose data and it will not be cleared by the *Clear highlights* action.

### Using event data in actions

When [a user interacts with the timeline](#interacting-with-the-timeline-component) (e.g. moving an item), the event can provide details about what changed. From DataMiner 10.3.0 CU14/10.4.0 CU2/10.4.5 onwards, you can use this event data in your actions.

This is useful because it allows you to:

- **Update your data through an Automation script** based on what the user did.

- **Keep the timeline and your data in sync**, so changes are not just visual.

For example, if a user resizes an item, the event gives you both the old and new time range, along with the item's details. You can then pass that information to a script that updates the actual data behind the timeline.

Each event provides the following data:

| Event | Exposed data |
|--|--|

> | Selecting a time range | *from* and *to* properties (both of type *Timespan*) |

Author: ```suggestion
| Selecting a time range | *from* and *to* properties (both of type *DateTime*) |
```

> | Selecting a time range | *from* and *to* properties (both of type *Timespan*) |

Author: ```suggestion
| Selecting a time range | The selected time range as *Timespan* objects. |
```
